search/api_test.py

```diff
@@ -1,365 +1,122 @@
-        EducationFactory.create(profile=program_enrollment.user.profile)
-        assert_search(es.search(), [program_enrollment])
-
-    def test_education_update(self):
-        """
-        Test that Education is reindexed after being updated
-        """
-        program_enrollment = ProgramEnrollmentFactory.create()
-        assert es.search()['total'] == 1
-        education = EducationFactory.create(profile=program_enrollment.user.profile)
-        education.school_city = 'city'
-        education.save()
-        assert_search(es.search(), [program_enrollment])
-
-    def test_education_delete(self):
-        """
-        Test that Education is removed from index after being deleted
-        """
-        program_enrollment = ProgramEnrollmentFactory.create()
-        education = EducationFactory.create(profile=program_enrollment.user.profile)
-        assert_search(es.search(), [program_enrollment])
-        education.delete()
-        assert_search(es.search(), [program_enrollment])
-
-    def test_employment_add(self):
-        """
-        Test that Employment is indexed after being added
-        """
-        program_enrollment = ProgramEnrollmentFactory.create()
-        assert es.search()['total'] == 1
-        EmploymentFactory.create(profile=program_enrollment.user.profile)
-        assert_search(es.search(), [program_enrollment])
+        cls.user = profile.user
+        cls.program = ProgramFactory.create()
+        Role.objects.create(
+            user=cls.user,
+            program=cls.program,
+            role=Staff.ROLE_ID
+        )
 
-    def test_employment_update(self):
+    def test_execute_search(self):  # pylint: disable=no-self-use
         """
-        Test that Employment is reindexed after being updated
+        Test that the execute_search method invokes the right method on the Search object
         """
-        program_enrollment = ProgramEnrollmentFactory.create()
-        assert es.search()['total'] == 1
-        employment = EmploymentFactory.create(profile=program_enrollment.user.profile)
-        employment.city = 'city'
-        employment.save()
-        assert_search(es.search(), [program_enrollment])
+        search_obj = Search()
+        search_obj.execute = Mock(name='execute')
+        execute_search(search_obj)
+        assert search_obj.execute.called
 
-    def test_employment_delete(self):
+    def test_create_search_obj_program_limit(self):
         """
-        Test that Employment is removed from index after being deleted
+        Test that Search objects are created with program-limiting query parameters
         """
-        program_enrollment = ProgramEnrollmentFactory.create()
-        employment = EmploymentFactory.create(profile=program_enrollment.user.profile)
-        assert_search(es.search(), [program_enrollment])
-        employment.delete()
-        assert_search(es.search(), [program_enrollment])
+        search_obj = create_search_obj(self.user)
```

Comment:
It doesn't look like there are any `create_search_obj` tests using either `search_param_dict` or `search_func`. Can you add tests or update existing tests to cover these parameters?